kylin-spark-project/kylin-spark-common/src/main/scala/org/apache/spark/sql/execution/datasource/FilePruner.scala

```diff
@@ -330,7 +330,7 @@ class FilePruner(cubeInstance: CubeInstance,
         }
       case _ =>
         //other unary filter like EqualTo, GreaterThan, GreaterThanOrEqual, etc.
-        if (filter.references.subsetOf(AttributeSet(col))) {
+        if (col != null && filter.references.subsetOf(AttributeSet(col))) {
```

Comment:
I think we can move the null-check to the beginning of the method, so that the mothed would not go into the recursion.